src/targets/stm32f1-generic/main.c

```diff
@@ -51,9 +56,33 @@ int main()
 
 	gpio_apply_config(gpio_config);
 
+	struct hid_hal_t hid_hal;
+	u8 info_functions[] = {
+		OI_FUNCTION_VERSION,
+		OI_FUNCTION_FW_INFO,
+		OI_FUNCTION_SUPPORTED_FUNCTION_PAGES,
+		OI_FUNCTION_SUPPORTED_FUNCTIONS,
+	};
+
+	/* create protocol config */
+	memset(&protocol_config, 0, sizeof(protocol_config));
+	protocol_config.device_name = "openinput stm32f1 Device";
```

Comment:
Let's keep this as `openinput Device` to match the name in the USB tables. AFAIK it should be the same, some tools will use this name, some tools will use the one from the table, it would be better to have it be consistent.

Reply:
Sure, makes sense, i took it from the uhid target that has i believe "openinput uhid device" and change uhid to stm32f1

Reply:
Right, that was implemented before we got the PID. Though thinking about it, I think we could still leverage this field, while keeping things somewhat consistent, by putting the target name in parenthesis.

```
openinput Device (Generic STM32F1)
```

```
openinput Device (Rival 310)
```

```
openinput Device (UHID virtual device)
```

Places that decide to use the USB table will get `openinput Device` and places that use the device string will have that extra information.